### ISSUE TITLE
Provide Type Declarations

### ISSUE THREAD

**Issue by timcole** [pulse-framework/pulse#66]
## Expected Behavior

Pulse should generate and provided type declarations on build to the npm package.

## Current Behavior

`ERROR: Could not find a declaration file for module 'pulse-framework'. '/Users/tim/Documents/(redacted)/node_modules/pulse-framework/dist/pulse.js' implicitly has an 'any' type.`

## Extra Information

Information can be found at http://www.typescriptlang.org/docs/handbook/declaration-files/publishing.html

Basically when you build for npm you should also generate type declarations for use in TypeScript.

This can be done using `tsc --declaration` and providing the declarations location in your `package.json` as `types`

## Environment Information
- Pulse Version: `2.2.5`

**Comment by TimothyCole** [comment]
It looks like if you wanna generate declarations via rollup it's not possible with the current typescript plugin you use https://github.com/rollup/plugins/issues/105 but it is possible with https://github.com/wessberg/rollup-plugin-ts

Also most of the of rollup plugins in https://github.com/pulse-framework/pulse/blob/421ef903ec805e30a987c305a7b713f6402c53b4/package.json#L33-L39 are no longer maintained for their `@rollup/x` format instead.
